src/nft/components/Box.ts

```diff
@@ -0,0 +1,47 @@
+import clsx, { ClassValue } from 'clsx'
+import * as React from 'react'
+import { animated } from 'react-spring'
+
+import { Atoms, atoms } from '../css/atoms'
+import { sprinkles } from '../css/sprinkles.css'
+
+type HTMLProperties<T = HTMLElement> = Omit<
+  React.AllHTMLAttributes<T>,
+  'as' | 'className' | 'color' | 'height' | 'width'
+>
+
+type Props = Atoms &
+  HTMLProperties & {
+    as?: React.ElementType
+    className?: ClassValue
+  }
+
+export const Box = React.forwardRef<HTMLElement, Props>(({ as = 'div', className, ...props }: Props, ref) => {
+  const atomProps: Record<string, unknown> = {}
+  const nativeProps: Record<string, unknown> = {}
+
+  for (const key in props) {
+    if (sprinkles.properties.has(key as keyof Omit<Atoms, 'reset'>)) {
```

Comment:
We need to omit 'reset' as it's not a sprinkle property and we get the following type error if it's not omitted
![Screen Shot 2022-08-15 at 15 21 36 ](https://user-images.githubusercontent.com/11512321/184729360-a5b282ac-97be-4b5f-acde-9c6483a41f01.png)

Reply:
You should just convert it directly to a sprinkles property, because that is what is being tested. I think ideally the types would look like this:

```
import { Sprinkles, sprinkles } from '../css/sprinkles.css'

type HTMLProperties<T = HTMLElement> = Omit<
  React.AllHTMLAttributes<T>,
  'as' | 'className' | 'color' | 'height' | 'width'
>

interface Props extends Atoms, HTMLProperties {
  as?: React.ElementType
  className?: ClassValue
}

export const Box = React.forwardRef<HTMLElement, Props>(({ as = 'div', className, ...props }: Props, ref) => {
  const atomProps: Record<string, unknown> = {}
  const nativeProps: Record<string, unknown> = {}

  for (const key in props) {
    if (sprinkles.properties.has(key as keyof Sprinkles)) {
      atomProps[key] = props[key as keyof Sprinkles]
    } else {
      nativeProps[key] = props[key as keyof HTMLProperties]
    }
  }
```

Reply:
If this breaks downstream usage, you should leave as-is.